Use configured hostname and port in CSP proxied values

cbliard opened the pull request:
The motivation is to be able to run the application locally in development on two different ports and hostnames: one running `dev` on port 3000, the other running `release/12.4` branch on port 3030. This was not possible because CSP hosts and ports for development were hardcoded.

cbliard committed:
Use configured hostname and port in CSP proxied values

This allows to use a different port when running the application
locally in development environment.

cbliard commented:
@b12f does this still work with the docker development setup?

dombesz commented on config/initializers/secure_headers.rb:
```diff
@@ -41,9 +41,8 @@
-      proxied = ['ws://localhost:3000', 'http://localhost:3000',
-                 'ws://localhost:4200', 'http://localhost:4200',
-                 FrontendAssetHelper.cli_proxy]
+      proxied = ["ws://#{Setting.host_name}", "http://#{Setting.host_name}",
```
I think we might need a default value for the `Setting.host_name`, because it can be set to an empty string via the admin interface.
<img width="1277" alt="image" src="https://user-images.githubusercontent.com/83396/208088311-3f6b1f9e-a15a-486f-91f3-3df413c7ac6c.png">

cbliard replied:
Good point.

It already has a [default value](https://github.com/opf/openproject/blob/633b05f495e239d4cfbc6fcf37c065626e61f9eb/config/constants/settings/definitions.rb#L441-L444). Maybe the admin should be changed to forbid empty host name, because it should be set to whatever value is used in the URL to access the application.

b12f commented:
I tested out this branch with a docker compose dev setup and it appears to work fine

cbliard committed:
Improve error message of #retry_block used in capybara tests

Add source line where the exception occured, remove useless output added by structured_warning gem, indicate when last try, add header and footer to better visualize the log message

cbliard committed:
Output log dir and disable pry by default

cbliard committed:
Fix flaky test

For some reason, the dropdown menu is not displayed after clicking the
dropdown. It has to be retried.
